web/src/main/java/ua/com/vertex/controllers/ViewCoursesController.java

```diff
@@ -0,0 +1,84 @@
+package ua.com.vertex.controllers;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.SessionAttributes;
+import org.springframework.web.servlet.ModelAndView;
+import ua.com.vertex.beans.Course;
+import ua.com.vertex.beans.User;
+import ua.com.vertex.logic.interfaces.CourseLogic;
+import ua.com.vertex.logic.interfaces.UserLogic;
+import ua.com.vertex.utils.DataNavigator;
+import ua.com.vertex.utils.EmailExtractor;
+
+import java.sql.SQLException;
+import java.util.List;
+
+@Controller
+@SessionAttributes(value = "viewCourses")
+@RequestMapping(value = "/viewCourses")
```

Comment:
In this case it is impossible, because of "@Target(value=METHOD)", not TYPE.
Link to docs: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/GetMapping.html

Reply:
agree